styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -232,9 +233,15 @@ private Response<TriggerRequest> triggerAndWait(TriggerRequest triggerRequest,
     } catch (Exception e) {
       return handleException(e);
     }
-
-    // todo: change payload to a struct returning the triggerId as well so the user can refer to it
-    return Response.forPayload(triggerRequest);
+    TriggerResponse response;
+    if (triggerRequest.triggerParameters().isPresent()) {
```

Comment:
would calling `TriggerResponse.of(triggerRequest.workflowId(), triggerRequest.parameter(),
       parameters, triggerId);` here be sufficient? Would a null `triggerParametres()` and `TriggerParameters.zero()` output the same?

Reply:
From the looks of the generated code for the automatter builder, passing `null` should behave the same way as not calling the setter on the builder (`Optional.empty()`), so something like this should work:

```java
TriggerResponse response =
    TriggerResponse.of(triggerRequest.workflowId(), triggerRequest.parameter(),
        triggerRequest.triggerParameters().orElse(null), triggerId);
``` 

If you tweaked the `of` factory method to accept `Optional<TriggerParameters>` you could even pass it directly through. I'm not a big fan of `Optional` for parameters though. I don't know. What do you think? :)

`TriggerParameters.zero()` is an internal thing and should probably not be passed either in or out, so better to go with `Optional.empty()`.

Reply:
I will try to, but otherwise will bring to the rest of the team. It's been a few years since I last worked on Styx 😬

Reply:
thanks!!

Reply:
`mvn test` at least runs fine on my machine without errors, but need to verify if that runs all the tests. Will try to look into that tomorrow.